Added parsing logic for the key=value body requests

KRoLer opened the pull request:
Added parsing logic for the requests of type **application/x-www-form-urlencoded** with the key=value body string. 

I didn't find another way to reach them except to extend an existing logic. 
If you think it will be useful I can update documentation for this feature. 
If you know another way how to do it I'll be also glad to hear it.

tranhungt commented:
@KRoLer Thanks for this. I think this could be useful in extending the applicability of this transformer, since we should support multiple request body types.
I can look more into the wiremock library to see if there are any other possible solutions.
We should go ahead and add the documentation to this PR.

Simon-Kaz commented:
Hey @KRoLer , I tried using your solution to parse an email, but I'm getting url encoded version back instead of the "@". Any way to disable the url encoding?

tranhungt commented on src/test/java/com/opentable/extension/BodyTransformerTest.java:
```diff
@@ -70,6 +70,58 @@ public void replaceNestedVariablesForXml() throws Exception {
+			.willReturn(aResponse()
+				.withStatus(200)
+				.withHeader("content-type", "application/json")
+				.withBody("{\"EmptyKey\":\"\"}")
```
If this is returning empty string, then we'd never know that it tried to parse the "EmptyKey" url body?

tranhungt commented on src/main/java/com/opentable/extension/BodyTransformer.java:
```diff
@@ -54,7 +56,20 @@ public ResponseDefinition transform(Request request, ResponseDefinition response
+                    for(String pair: pairedValues)
+                    {
+                        String[] values = pair.split("=");
+                        object.put(values[0], values.length > 1 ? values[1] : "");
```
Shouldn't we `URLDecoder.decode` the values[1]?

tranhungt commented:
@KRoLer Sorry for the late review. Could you check the comments, fix the conflict, and add some documentation? I can merge asap.

Thanks.

KRoLer commented:
@tranhungt, thanks for the review. I've fixed issue for the case when it has only parameter name and commits it; I've not planned to change my PR. If you accept this solution at all, I'll refactor and prettify code before the merge. :-)

KRoLer commented:
Hey, @tranhungt,

I've updated my PR. 
Now it had an actual code version, fixed test issues and added all needed logic, including UTF-8 values decoding.

tranhungt commented:
Thanks @KRoLer this looks great! I'll check it out tomorrow morning.
